src/main/java/org/spongepowered/api/ai/goal/agent/LookIdleGoal.java

```diff
@@ -29,18 +29,18 @@
 import org.spongepowered.api.ai.goal.GoalBuilder;
 import org.spongepowered.api.entity.living.Agent;
 
-public interface LookIdleGoal extends Goal<Agent> {
+public interface LookIdleGoal<O extends Agent> extends Goal<O> {
 
     /**
      * Creates a new {@link Builder} to build a new {@link LookIdleGoal}.
      *
      * @return A new builder
      */
-    static Builder builder() {
+    static <O extends Agent, B extends Builder<O, LookIdleGoal<O>, B>> B builder(O type) {
         return Sponge.getRegistry().createBuilder(Builder.class);
```

Comment:
if you're using the raw type, is there a raw type cast or unchecked cast going on?

Reply:
The use of the `Builder` must be a raw type without generics because the generic self type can never be specified. When users need to use, they need another method in which this generic type can be specified again for usage.